drivers/spi/spi_ll_stm32.c

```diff
@@ -53,10 +56,36 @@ LOG_MODULE_REGISTER(spi_ll_stm32);
 
 
 #ifdef CONFIG_SPI_STM32_DMA
-/* dummy value used for transferring NOP when tx buf is null
- * and use as dummy sink for when rx buf is null
- */
-uint32_t dummy_rx_tx_buffer;
+
+#if CONFIG_SPI_STM32_UNALIGNED_DMA
+
+#define SPI_STM32_ROUND_UP(x) ROUND_UP((uint32_t)(x), __SCB_DCACHE_LINE_SIZE)
+#define SPI_STM32_ROUND_DOWN(x) \
+			ROUND_DOWN((uint32_t)(x), __SCB_DCACHE_LINE_SIZE)
+
+void spi_stm32_flush_cache(const void *ptr, uint32_t size)
```

Comment:
In practice both `spi_stm32_flush_cache` and `spi_stm32_invalidate_cache` are called with aligned address and length that are a multiple of `__SCB_DCACHE_LINE_SIZE`, so there is no need for all that alignment around.

Reply:
Yeah those checks were from before I realized I could not just invalidate a few bytes by finding the cacheline and invalidating the that line. The rounding is done again in the actual ARM code, so these checks could go.  Maybe a ASSERT could be put in place to make it very clear the address and size must be aligned ?

Reply:
The ASSERT is a really good idea.